document/cloud-client/analyze_form.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def analyze_form(project_id, input_uri):
+    """Parse a form"""
+    # [START analyze_form]
+    from google.cloud import documentai_v1beta2 as documentai
+    from google.cloud.documentai_v1beta2 import types
+
+    client = documentai.DocumentUnderstandingServiceClient()
+
+    gcs_source = types.GcsSource(uri=input_uri)
+    input_config = types.InputConfig(
+        gcs_source=gcs_source, mime_type='application/pdf')
+
+    # Improve form parsing results by providing key-value pair hints.
+    # For each key hint, key is text that is likely to appear in the
+    # document as a form field name (i.e. "DOB").
+    # Value types are optional, but can be one or more of:
+    # ADDRESS, LOCATION, ORGANIZATION, PERSON, PHONE_NUMBER, ID,
+    # NUMBER, EMAIL, PRICE, TERMS, DATE, NAME
+    key_value_pair_hints = [
+        types.KeyValuePairHint(key='Name', value_types=['NAME']),
+        types.KeyValuePairHint(key='Agreement Number', value_types=['NUMBER'])
+    ]
+
+    # Setting enabled=True enables form extraction
```

Comment:
What is form extraction? Using these `key_value_pair_hints` or something else

Reply:
Not sure what you mean. This API is kind of weird. To analyze a doc, you call a generic `PrrocessDocumentRequest` method. To do specifically form extraction, you must pass `documentai.types.FormExtractionParams(enabled=True)`. You can optionally also add the `key_value_pair_hints`.

Reply:
What makes `form_extraction` different from the default response? is I guess where I'm confused.

Reply:
Form extraction and key/value pairs hints are linked, IIUC. 

Say you have a form that you want to analyze that looks like this:

```
TO: Jane Doe
ADDRESS: 12345 Main St.
TOTAL: $100.00
INVOICE NUMBER: 12345
```
Where you have defined fields (`TO`, `ADDRESS`, `TOTAL`, etc) that expect a specific type. In your `form_extraction_params` config, you can provide a series of keys for the service to look for along with an expected data type. The helps bias the recognizer towards these entities (and their values) in the PDF.

I think.

Reply:
> Form extraction and key/value pairs hints are linked, IIUC.

Can we get this confirmed, if the use of `key_value_pairs` is tied to `form_extraction` or if they are independent?

Mostly I'm wondering, if you pass the created `params` object below, why does it also need a `enabled=True` field. Shouldn't that be implied when you pass the object already or is it a required object that could be empty?